discovery-infra/test_infra/utils/global_variables/global_variables.py

```diff
@@ -53,12 +52,25 @@ def __post_init__(self):
                 client=self.get_api_client()
         self._set("openshift_version", utils.get_openshift_version(allow_default=True, client=client))
 
-        for (env, expected), values in _triggers.items():
-            if getattr(self, env) == expected:
-                for k, v in values.items():
-                    self._set(k, v)
-                self._triggered.append(env)
-                log.info(f"{env.upper()} is triggered. Updating global variables: {values}")
+        for conditions, values in _triggers.items():
+            # A dictionary doesn't support lists so the only way to verify
+            # multi-level is by checking the inner value is also a tuple
+            if isinstance(conditions[0], tuple):
+                if all(map(lambda condition: self.is_set(condition[0], condition[1]), conditions)):
+                    self._handle_trigger(conditions, values)
+            else:
+                if self.is_set(conditions[0], conditions[1]):
+                    self._handle_trigger(conditions, values)
+
+    def is_set(self, var, expected):
+        return getattr(self, var) == expected
+
+    def _handle_trigger(self, conditions, values):
+        for k, v in values.items():
+            self._set(k, v)
+
+        self._triggered.append(conditions)
+        log.info(f"{conditions} is triggered. Updating global variables: {values}")
```

Comment:
It took me a while to understand what happens here
Your changes are better than the previous code, so I don't think more than that is needed to be done in the scope of the current PR